tools/configure.c

```diff
@@ -1590,6 +1590,48 @@ static void refresh(void)
     }
 }
 
+static void generate_info()
+{
+  /* Append the configuration information to the .info file */
+
+  FILE *stream;
+  char *tmp_boarddir;
+
+  snprintf(g_buffer, BUFFER_SIZE, "%s%c%s", g_topdir, g_delim, ".info");
+  stream = fopen(g_buffer, "w");
+  if (!stream)
+    {
+      fprintf(stderr,
+              "ERROR: Failed to open %s for write mode: %s\n",
+              g_buffer, strerror(errno));
+      exit(EXIT_FAILURE);
+    }
+
+  tmp_boarddir = double_appdir_backslashes(g_boarddir);
+  if (NULL == tmp_boarddir)
+    {
+      fprintf(stderr,
+              "ERROR: Failed to double boarddir backslashes\n");
+      exit(EXIT_FAILURE);
+    }
+
+  fprintf(stream, "CONFIG_BASE_DEFCONFIG=\"%s\"\n", tmp_boarddir);
```

Comment:
configure.sh already appends settings to .config.  It is a standard practice to do this.  You would need to have a definition in a Kconfig file, but only to prevent the appended setting from being lost if re-configured.  Like:

    config BASE_DEFCONFIG
        string
        default ""

It does not have to have a value.

Reply:
Then I'll fuse both ways that I've implemented this to add this info to `config.h` directly.

Reply:
Doesn't it make sense to provide a `.info` containing not only the `CONFIG_BASE_DEFCONFIG` but also the diff between it and the current config? That would be very useful to check which configs were removed or added from a specific `defconfig`.

Reply:
> But "make savedefconfig" will update defconfig to reflect your change in .config.

Yes, but the `.info` would be used to be parsed by `nxdiag`. Thus, it'd need to get it from an already "ready-to-be-parsed" file. The current `.config` may be already "patched" and `make savedefconfig` would provide the `defconfig`. But how would the `nxdiag` run such a command while being built?

Reply:
@patacongo I tried to create the config as you suggested but the value is always the default. If I make it a named config, it works as expected. Any idea why ?
```
config BASE_DEFCONFIG
    string
    default ""
```
Generates:
```
#
# Build Configuration
#
CONFIG_APPS_DIR="../apps"
CONFIG_BASE_DEFCONFIG=""
CONFIG_BUILD_FLAT=y
# CONFIG_BUILD_2PASS is not set
```

But,
```
config BASE_DEFCONFIG
    string "Base defconfig"
    default ""
```
Generates
```
#
# Build Configuration
#
CONFIG_APPS_DIR="../apps"
CONFIG_BASE_DEFCONFIG="esp32-devkitc:nsh"
CONFIG_BUILD_FLAT=y
# CONFIG_BUILD_2PASS is not set
```

Reply:
I've updated the MR with your suggestions